components/Publish.vue

```diff
@@ -118,29 +121,65 @@ export default {
   props: ["ad", "provider", "contract", "ketherNFT", "showNSFW"],
   data() {
     return {
+      canPublish: true,
+      warnings: [],
       inProgress: false,
       error: null,
     }
   },
+  watch: {
+    ad() {
+      // Reset checkImage
+      this.canPublish = true;
+      this.warnings = [];
+    },
+  },
   methods: {
+    async checkImage() {
+      // FIXME: This is hacky AF for now
+      const warnings = [];
+      const el = this.$refs.adPreview.$el.children[0];
+      const maxWidth = this.ad.width * 20;
+      const maxHeight = this.ad.height * 20;
+      const maxRes = maxWidth * maxHeight * 2;
+
+      if (el.naturalWidth > maxWidth) {
+        warnings.push(`Image is too wide: Width is ${el.naturalWidth}px but it should be ${maxWidth / 2}px (or ${maxWidth}px for HiDPI).`);
+      }
+      if (el.naturalHeight > maxHeight) {
+        warnings.push(`Image is too tall: Height is ${el.naturalHeight}px but it should be ${maxHeight / 2}px (or ${maxHeight}px for HiDPI).`);
+      }
+      if ((el.naturalWidth / el.naturalHeight) != (maxWidth / maxHeight)) {
+        warnings.push(`Image aspect ratio is incorrect, it will appear squished and blurry.`);
+      }
+      const resTooBig = (el.naturalWidth * el.naturalHeight > maxRes);
+      if (resTooBig) {
+        warnings.push(`Image is way too big. Please resize to ${maxWidth}x${maxHeight}px or smaller before publishing.`);
+      }
+      this.canPublish = !resTooBig;
+      this.warnings = warnings;
+    },
     async publish() {
+      await this.checkImage();
+      if (!this.canPublish) return;
+      const ad = this.ad;
       const signer = await this.provider.getSigner();
       const signerAddr = await signer.getAddress();
-      if (signerAddr.toLowerCase() != this.ad.owner) {
-        this.error = 'Incorrect active wallet. Must publish with: ' + this.ad.owner;
+      if (signerAddr.toLowerCase() != ad.owner) {
```

Comment:
Might as well
```suggestion
      if (!$address.equal(signerAddr, ad.owner)) {
```

Reply:
Might as well add a change that breaks things? :P 

It's `this.$address` I think, wanna just do all of them in one pass?

Reply:
hahaha yeah sure :)